repository/src/main/java/uk/gov/hmcts/ccd/definition/store/repository/entity/CaseFieldEntity.java

```diff
@@ -213,6 +214,35 @@ public boolean isMetadataField() {
         return dataFieldType == DataFieldType.METADATA;
     }
 
+    @Transient
+    public FieldEntity findNestedElementByPath(String path) {
```

Comment:
Can we try to make this return Optional<FieldEntity> and also not throw exceptions in both this and the child method so that the callers can decide based on the absence? I am reusing this logic for markdown for collection items story and have the caller.

```
    @Transient
    public Optional<FieldEntity> findNestedElementByPath(String path) {
        if (StringUtils.isBlank(path)) {
            return Optional.of(this);
        }
        if (this.getFieldType().getChildren().isEmpty()) {
            Optional.empty();
        }
        List<String> pathElements = Arrays.stream(path.trim().split("\\.")).collect(Collectors.toList());

        return reduce(this.getFieldType().getChildren(), pathElements);
    }

    private Optional<FieldEntity> reduce(List<ComplexFieldEntity> caseFields, List<String> pathElements) {
        String firstPathElement = pathElements.get(0);

        Optional<FieldEntity> caseField = caseFields.stream()
                                                    .filter(e -> e.getReference().equals(firstPathElement))
                                                    .map(e -> (FieldEntity)e)
                                                    .findFirst();

        if (!caseField.isPresent()) {
            return Optional.empty();
        }

        if (pathElements.size() == 1) {
            return caseField;
        } else {
            List<ComplexFieldEntity> complexFieldEntities = caseField.get().getFieldType().getChildren();
            List<String> tail = pathElements.subList(1, pathElements.size());

            return reduce(complexFieldEntities, tail);
        }
    }
```